src/main/java/nl/novi/techiteasy/config/SpringSecurityConfig.java

```diff
@@ -0,0 +1,78 @@
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+public class SpringSecurityConfig {
+
+    private final CustomUserDetailsService customUserDetailsService;
+    private final JwtRequestFilter jwtRequestFilter;
+
+    public SpringSecurityConfig(CustomUserDetailsService customUserDetailsService, JwtRequestFilter jwtRequestFilter) {
+        this.customUserDetailsService = customUserDetailsService;
+        this.jwtRequestFilter = jwtRequestFilter;
+    }
+
+    // PasswordEncoderBean. Deze kun je overal in je applicatie injecteren waar nodig.
+    // Je kunt dit ook in een aparte configuratie klasse zetten.
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+
+    // Authenticatie met customUserDetailsService en passwordEncoder
+    @Bean
+    public AuthenticationManager authenticationManager(HttpSecurity http, PasswordEncoder passwordEncoder) throws Exception {
+        var auth = new DaoAuthenticationProvider();
+        auth.setPasswordEncoder(passwordEncoder);
+        auth.setUserDetailsService(customUserDetailsService);
+        return new ProviderManager(auth);
+    }
+
+
+
+    // Authorizatie met jwt
+    @Bean
+    protected SecurityFilterChain filter (HttpSecurity http) throws Exception {
+
+        //JWT token authentication
+        http
+                .csrf(csrf -> csrf.disable())
+                .httpBasic(basic -> basic.disable())
+                .cors(Customizer.withDefaults())
+                .authorizeHttpRequests(auth ->
+                                auth
+                                        // Wanneer je deze uncomments, staat je hele security open. Je hebt dan alleen nog een jwt nodig.
+//                .requestMatchers("/**").permitAll()
+                                        .requestMatchers(HttpMethod.POST, "/users").hasRole("ADMIN")
+                                        .requestMatchers(HttpMethod.GET,"/users").hasRole("USER")
+                                        .requestMatchers(HttpMethod.POST,"/users/**").hasRole("USER")
+                                        .requestMatchers(HttpMethod.DELETE, "/users/**").hasRole("ADMIN")
+                                        .requestMatchers(HttpMethod.PUT, "/users").hasRole("USER")
+                                        /*TODO voeg de antmatchers toe voor admin(post en delete) en user (overige)*/
```

Comment:
Ik neem aan dat je hier overheen hebt gekeken. Misschien is het handig om te kijken of je een tv kan aanmaken via een post request of kijken of het verwijderen werkt